Remotion/Data/DomainObjects/ConfigurationLoader/ReflectionBasedConfigurationLoader/LegacyPropertyDefaultValueProvider.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Runtime.Serialization;
+using Remotion.Data.DomainObjects.Mapping;
+using Remotion.ExtensibleEnums;
+using Remotion.Reflection;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.ConfigurationLoader.ReflectionBasedConfigurationLoader
+{
+  /// <summary>
+  /// Implements the legacy behavior for <see cref="PropertyDefinition.DefaultValue"/>, which sets non-nullable reference-<see cref="Type"/> properties to
+  /// non-<see langword="null"/> values. This affects all <see cref="Array"/>, <see cref="string"/>, and <see cref="ExtensibleEnum{T}"/> properties.
+  /// </summary>
+  /// <threadsafety static="true" instance="true" />
+  public class LegacyPropertyDefaultValueProvider : IPropertyDefaultValueProvider
+  {
+    private static readonly ConcurrentDictionary<Type, IExtensibleEnum> s_uninitializedValues = new();
+    private static readonly object?[] s_extensibleEnumValues = { "Uninitialized" };
```

Comment:
Are we still missing the mapping validation rules, or is the following pre-existing test actually covering this scenario and if so, how was this doable before our rework?
see Remotion.Data.DomainObjects.UnitTests.Mapping.Validation.Logical.MandatoryExtensibleEnumTypeHasValuesDefinedValidationRuleTest
and NotSupportedPropertyTypesIntegrationTests.PropertyTypeOfExtensibleEnum_WithoutValues_Mandatory

Reply:
The mapping validation was always there, so the existing tests cover that it works.
Before extracting `LegacyPropertyDefaultValueProvider`, a call to `PropertyDefinition.DefaultValue` would throw an exception if there were no defined values; this was okay because calls to that property would have only occurred after the mapping validation had already found the problem and thrown a better exception.
Since we now inject the default value into `PropertyDefinition` at creation, that timing has changed; we no longer want to throw an exception, because that would preempt the mapping validation.

Reply:
Good, same conclusion I had. Original validation covers everything, we just needed to prevent the exception during initialization of the object graph. No extra work. all good.